driver/health/event_manager_default.go

```diff
@@ -0,0 +1,51 @@
+package health
+
+import (
+	"context"
+	"errors"
+	"reflect"
+)
+
+type DefaultHealthEventManager struct {
+	evtChan                chan interface{}
+	listenerEventTypeCache map[reflect.Type]Readiness
+}
+
+func NewDefaultHealthEventManager() *DefaultHealthEventManager {
+	return &DefaultHealthEventManager{
+		evtChan:                make(chan interface{}),
+		listenerEventTypeCache: make(map[reflect.Type]Readiness),
+	}
+}
+
+func (h *DefaultHealthEventManager) Dispatch(event interface{}) {
+	select {
+	case h.evtChan <- event:
+	default:
+	}
+}
+
+func (h *DefaultHealthEventManager) AddListener(listener Readiness) error {
+	for _, evtType := range listener.EventTypes() {
+		evtTypeVal := reflect.TypeOf(evtType)
+		if _, ok := h.listenerEventTypeCache[evtTypeVal]; ok {
+			return errors.New("event type already registered")
```

Comment:
It is better to export the error and return 
```suggestion
			return errors.WithStack(ErrEventTypeAlreadyRegistered)
```